backend/app/Services/LocationFetcher.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Internships\Services;
+
+use Illuminate\Http\Client\PendingRequest;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Log;
+
+class LocationFetcher
+{
+    protected const API_URL = "https://api.mapbox.com/geocoding/v5/mapbox.places/";
+
+    protected string $mapboxToken;
+    protected PendingRequest $request;
+    protected Collection $receivedData;
+
+    public function __construct()
+    {
+        $this->mapboxToken = env("MAPBOX_TOKEN");
+        $this->request = Http::withUserAgent(env("APP_NAME", "Internships"));
+    }
+
+    public function query(string $address): self
+    {
+        $addressAsUrl = urlencode($address);
+
+        $response = $this->request->get(self::API_URL . $addressAsUrl . ".json", [
+            "access_token" => $this->mapboxToken,
+        ]);
+
+        $this->receivedData = collect($response->json());
+        if ($this->receivedData->has("message")) {
+            Log::warning("Mapbox response contains message: " . $this->getMessage());
+        }
+
+        return $this;
+    }
+
+    public function get(): Collection
```

Comment:
Let's think about better name for this function.

Reply:
Well, I guess getLocations could do.